backend/routes/follows.js

```diff
@@ -4,56 +4,113 @@ import { Stock } from "../models/stock.js";
 
 const router = express.Router();
 
+const checkSession = (req, res, next) => {
+    const user = req.session.user;
+    if (!user) {
+        res.status(401).json({ error: "Missing Session." });
+    } else {
+        next();
+    }
+};
+
+router.use(checkSession);
+
 router.get("/follows", async (req, res) => {
     const user = req.session.user;
 
-    const follows = await Follow.findAll({
-        where: { UserId: user.id },
-        include: [{ model: Stock }]
-    });
+    try {
+        const follows = await Follow.findAll({
+            where: { UserId: user.id },
+            include: [{ model: Stock }]
+        });
+
+        if (follows.length === 0) {
+            return res
+                .status(404)
+                .json({ error: "This user doesn't follow any stocks." });
+        }
 
-    const stocks = follows.map((follow) => follow.Stock);
+        const stocks = follows.map((follow) => follow.Stock);
 
-    res.status(200).json({ stocks });
+        res.status(200).json({ stocks });
+    } catch (error) {
+        res.status(500).json({ error });
+    }
 });
 
 router.post("/follow", async (req, res) => {
-    const ticker = req.body.ticker?.toUpperCase();
-
     const user = req.session.user;
-    const stock = await Stock.findOne({ where: { ticker } });
+    const ticker = req.body.ticker?.toUpperCase();
 
-    const followData = {
-        UserId: user.id,
-        StockId: stock.id
-    };
+    if (!ticker) {
+        return res.status(400).json({ error: "Invalid request body." });
+    }
 
     try {
-        const follow = await Follow.create(followData);
-        res.status(200).json({ follow });
+        const stock = await Stock.findOne({ where: { ticker } });
+
+        if (stock === null) {
+            return res
+                .status(404)
+                .json({ error: "This stock does not exist in database." });
+        }
+
+        const followData = {
+            UserId: user.id,
+            StockId: stock.id
+        };
+
+        const follow = await Follow.findOne({ where: followData });
+
+        if (follow != null) {
```

Comment:
on line 52 you used `=== null` and on 65 used `!= null`. Would be good to be more consistent and be sure to understand the difference between `==` and `===` (or `!=` and `!==`) in JS -- [here's](https://stackoverflow.com/questions/38977829/what-is-the-difference-between-null-and-null) a quick explainer to understand the difference